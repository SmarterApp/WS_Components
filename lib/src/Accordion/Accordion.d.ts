import * as React from "react";
export interface AccordionProps {
    accordionTitle: string;
    isOpen: boolean;
    toggleExpand: () => void;
}
export declare class Accordion extends React.Component<AccordionProps, {}> {
    constructor(props: AccordionProps);
    renderContent(isOpen: boolean): JSX.Element | undefined;
    renderCarat(isOpen: boolean): JSX.Element | undefined;
    render(): JSX.Element;
}
