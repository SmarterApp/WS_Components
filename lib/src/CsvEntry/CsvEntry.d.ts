import * as React from "react";
import { ItemRevisionModel, CsvRowModel } from "@src/index";
export interface CsvEntryProps {
    onItemsUpdate: (items: ItemRevisionModel[]) => void;
    onBlur: () => void;
}
export interface CsvEntryState {
    csvInputValue?: string;
    csvData: CsvRowModel[];
}
export declare class CsvEntry extends React.Component<CsvEntryProps, CsvEntryState> {
    constructor(props: CsvEntryProps);
    renderHelpButton(): JSX.Element;
    handleCsvChange(event: React.FormEvent<HTMLTextAreaElement>): void;
    handleCsvBlur: () => void;
    render(): JSX.Element;
}
