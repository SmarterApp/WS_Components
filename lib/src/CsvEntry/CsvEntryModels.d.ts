import { ItemRevisionModel } from "../ItemBank/ItemBankModels";
export interface CsvRowModel extends ItemRevisionModel {
    index: number;
}
export declare function parseCsv(csvValue?: string): CsvRowModel[];
