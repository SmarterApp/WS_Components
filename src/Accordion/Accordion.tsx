import * as React from "react";

export interface AccordionProps {
  accordionTitle: string;
  contentItem: JSX.Element;
}

export interface AccordionState {
  title: string;
  isOpen: boolean;
}

export class Accordion extends React.Component<AccordionProps, AccordionState> {
  constructor(props: AccordionProps) {
    super(props);

    this.state = { title: this.props.accordionTitle, isOpen: false };
  }

  handleShowContent = () => {
    this.setState({ isOpen: !this.state.isOpen });
  };

  renderContent(isOpen: boolean): JSX.Element | undefined {
    let content: JSX.Element | undefined;

    if (isOpen) {
      content = (
        <div className="accordion-content">{this.props.contentItem}</div>
      );
    }

    return content;
  }

  renderCarat(isOpen: boolean): JSX.Element | undefined {
    let content: JSX.Element | undefined;

    if (isOpen) {
      content = <i className="fa fa-chevron-up carat" />;
    } else {
      content = <i className="fa fa-chevron-down carat" />;
    }

    return content;
  }

  render() {
    return (
      <div className="accordion-container">
        <div className="accordion-bar" onClick={this.handleShowContent}>
          <div className="accordion-content">
            {this.state.title}
            {this.renderCarat(this.state.isOpen)}
          </div>
        </div>
        {this.renderContent(this.state.isOpen)}
      </div>
    );
  }
}
