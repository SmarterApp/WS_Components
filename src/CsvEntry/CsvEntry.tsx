import * as React from "react";
import {
  generateTooltip,
  ItemRevisionModel,
  CsvRowModel,
  NamespaceModel,
  parseCsv,
  Accordion
} from "@src/index";

export interface CsvEntryProps {
  namespaces: NamespaceModel[];
  onItemsUpdate: (items: ItemRevisionModel[]) => void;
  onBlur: () => void;
}

export interface CsvEntryState {
  csvInputValue?: string;
  csvData: CsvRowModel[];
}

export class CsvEntry extends React.Component<CsvEntryProps, CsvEntryState> {
  constructor(props: CsvEntryProps) {
    super(props);

    this.state = { csvData: [] };
  }

  renderHelpButton() {
    const helpText: JSX.Element = (
      <p>Enter namespace, bank key, item id, and section for the items you would like to add</p>
    );
    const displayText: JSX.Element = (
      <button
        className="item-nav-btn btn btn-default btn-sm about-item-btn"
        role="button"
        aria-label="Open help text"
      >
        <span className="fa fa-info-circle" aria-hidden="true" />
        Help
      </button>
    );

    return (
      <div className="help-button">
        {generateTooltip({
          helpText,
          displayText,
          displayIcon: undefined
        })}
      </div>
    );
  }

  handleCsvChange(event: React.FormEvent<HTMLTextAreaElement>) {
    const rawCsv: string = event.currentTarget.value;

    this.setState({
      csvInputValue: rawCsv
    });
  }

  handleCsvBlur = () => {
    const { csvInputValue } = this.state;
    const { namespaces } = this.props;
    const csvData = parseCsv(csvInputValue, namespaces);

    this.setState({
      csvData
    });
    this.props.onItemsUpdate(csvData);
    this.props.onBlur();
  };

  render() {
    return (
      <div className="csv-entry-wrapper">
        {this.renderHelpButton()}
        <textarea
          className="csv-text-entry"
          onChange={e => this.handleCsvChange(e)}
          value={this.state.csvInputValue}
          onBlur={this.handleCsvBlur}
        />
      </div>
    );
  }
}
