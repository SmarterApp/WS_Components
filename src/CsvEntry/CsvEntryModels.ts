import { ItemRevisionModel } from "../ItemBank/ItemBankModels";

export interface CsvRowModel extends ItemRevisionModel {
  index: number;
}

export function parseCsv(csvValue: string | undefined) {
  const data: CsvRowModel[] = [];

  if (csvValue) {
    const lines = csvValue.split("\n");
    let index = 0;

    for (const line of lines) {
      const values = line.split(",");

      const row: CsvRowModel = {
        index,
        itemKey: +values[1],
        bankKey: +values[0],
        section: values[2]
      };

      data.push(row);
      index = index + 1;
    }
  }

  return data;
}
