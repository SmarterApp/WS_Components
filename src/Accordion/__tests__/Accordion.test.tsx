import * as React from "react";
import { shallow } from "enzyme";
import { Accordion, AccordionProps, AccordionState } from "../Accordion";
import { CsvEntry } from "../../CsvEntry/CsvEntry";

function getJSXElement() {
  return <div>test</div>;
}
const accordionProps: AccordionProps = {
  accordionTitle: "testAccordion",
  contentItem: getJSXElement()
};

const accordionstate: AccordionState = {
  title: "testAccordion",
  isOpen: true
};

describe("ItemBankEntry", () => {
  const wrapper = shallow(<Accordion {...accordionProps} />);

  it("matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("renders open accordion", () => {
    const wrapperInstance = wrapper.instance() as Accordion;
    wrapperInstance.setState({ isOpen: accordionstate.isOpen });

    expect(wrapper).toMatchSnapshot();
  });

  it("call handle show content", () => {
    const wrapperInstance = wrapper.instance() as Accordion;
    wrapperInstance.handleShowContent();
    expect(wrapper).toMatchSnapshot();
  });
});
