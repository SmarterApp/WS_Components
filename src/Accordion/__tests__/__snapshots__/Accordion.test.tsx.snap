// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemBankEntry call handle show content 1`] = `
<div
  className="accordion-container"
>
  <div
    className="accordion-bar"
    onClick={[Function]}
    role="button"
  >
    <div
      className="accordion-content"
    >
      testAccordion
      <span
        aria-hidden="true"
        className="fa fa-chevron-down carat"
      />
    </div>
  </div>
</div>
`;

exports[`ItemBankEntry matches snapshot 1`] = `
<div
  className="accordion-container"
>
  <div
    className="accordion-bar"
    onClick={[Function]}
    role="button"
  >
    <div
      className="accordion-content"
    >
      testAccordion
      <span
        aria-hidden="true"
        className="fa fa-chevron-down carat"
      />
    </div>
  </div>
</div>
`;

exports[`ItemBankEntry renders open accordion 1`] = `
<div
  className="accordion-container"
>
  <div
    className="accordion-bar"
    onClick={[Function]}
    role="button"
  >
    <div
      className="accordion-content"
    >
      testAccordion
      <span
        aria-hidden="true"
        className="fa fa-chevron-down carat"
      />
    </div>
  </div>
</div>
`;
